influxql/ast.go

```diff
@@ -2112,6 +2116,20 @@ func (s *SelectStatement) validateAggregates(tr targetRequirement) error {
 	return nil
 }
 
+// validateFill ensures that the fill option matches the query type.
+func (s *SelectStatement) validateFill() error {
+	info := newSelectInfo(s)
+	if len(info.calls) == 0 {
+		switch s.Fill {
```

Comment:
Yea, `fill(null)` is necessary anyway since that's the default, but it's also possible since `fill(null)` will always produce a point and doesn't require looking at future points. Same with `fill(previous)`. Both of them will give valid values. `fill(none)` is problematic because it doesn't produce any point (which causes the iterators to become unaligned) and `fill(linear)` requires reading a future point which auxiliary iterators can't do.